core/src/Cabana_LinkedCellList.hpp

```diff
@@ -319,6 +319,87 @@ void permute(
     permute( linked_cell_list.binningData(), aosoa );
 }
 
+//---------------------------------------------------------------------------//
+/*!
+  \brief Given a linked cell list and range permute part of an AoSoA.
+
+  \tparam LinkedCellListType The linked cell list type.
+
+  \tparm AoSoA_t The AoSoA type.
+
+  \param linked_cell_list The linked cell list to permute the AoSoA with.
+
+  \param aosoa The AoSoA to permute.
+
+  \param begin The beginning index of the AoSoA range to sort.
+
+  \param end The end index of the AoSoA range to sort.
+*/
+template<class LinkedCellListType, class AoSoA_t>
+void permute(
+    const LinkedCellListType& linked_cell_list,
+    AoSoA_t& aosoa,
+    const std::size_t begin,
+    const std::size_t end,
+    typename std::enable_if<(is_linked_cell_list<LinkedCellListType>::value &&
+                             is_aosoa<AoSoA_t>::value),
+    int>::type * = 0 )
```

Comment:
@sslattery  You guys are overusing/misusing SFINAE.  This slows down compile times and does not bring you much.  The error is still going to be about the compiler not finding a matching function to call just as without the template parameters.  Consider using static assertions instead.

Reply:
yeah this is a good point here. in some cases this is necessary to get the compiler to pick the right function implementation but in this case a static assert is more appropriate.

Reply:
@streeve In these functions you can just delete the `enable_if` lines and replace them with `static_assert` in the function implementation using the same boolean check in the `enable_if`

Reply:
Other than that this API is good

Reply:
I've left the ``enable_if`` since the compiler wasn't able to distinguish the aosoa and slice variants